salt/_modules/metalk8s_drain.py

```diff
@@ -0,0 +1,512 @@
+        #   2) by checking err/warning message and take action?
+        #   3) when reponse.items is empty?
+        #   Note: taking option 3...
+        response = self.get_controller(pod.metadata.namespace, controller_ref)
+        if self.not_found(response):
+            raise DrainException(
+                "Missing pod controller for '{0}'".format(controller_ref.name)
+            )
+        return controller_ref
+
+    def daemonset_filter(self, pod):
+        '''Check if given pod can be elected for deletion daemonset-wise.
+
+        A daemonset will generally not be deleted, regardless of flags, but we
+        can determine if their presence constitutes an error.
+
+        The exception to this rule is when the pod is orphaned, meaning that
+        the management resource is not found, and we have the `force` flag set.
+
+        return: (deletable: bool, warn: str)
+        raises: DrainException if daemonset is in unexpected state
+        '''
+        controller_ref = _get_controller_of(pod)
+
+        if controller_ref is None or controller_ref.kind != "DaemonSet":
+            return True, ""
+
+        controller = self.get_controller(pod.metadata.namespace, controller_ref)
+
+        if self.not_found(controller):
+            if self._force:
+                # Not found and forcing: remove orphan pods with warning
+                return (
+                    True,
+                    "Default to deletable on pod with no controller found"
+                )
+
+            raise DrainException(
+                "Missing pod controller for '{0}'".format(controller_ref.name)
+            )
+
+        if not self._ignore_daemonset:
+            raise DrainException(self.FATAL_MSG["daemonset"])
+
+        return False, self.WARNING_MSG["daemonset"]
+
+    def get_pods_for_eviction(self):
+        '''Get list of pods that can be deleted/evicted on
+        node identified by the name `node_name`.
+        '''
+        warnings = {}
+        failures = {}
+        pods = []
+
+        try:
+            api_instance = kubernetes.client.CoreV1Api()
+            api_response = api_instance.list_pod_for_all_namespaces(
+                field_selector='spec.nodeName={0}'.format(self._node_name)
+            )
+        except (ApiException, HTTPError) as exc:
+            if isinstance(exc, ApiException) and exc.status == 404:
+                return []
+            log.exception(
+                'Exception when calling '
+                'CoreV1Api->list_pod_for_all_namespaces')
+            raise CommandExecutionError(exc)
+
+        for pod in api_response.items:
+            pod_deletable = True
+            for filt in (
+                    _mirrorpod_filter,
+                    self.localstorage_filter,
+                    self.unreplicated_filter,
+                    self.daemonset_filter
+            ):
+                try:
+                    filter_deletable, warning = filt(pod)
```

Comment:
As per our Python Best Practices guide, boolean variables should use a prefix like `is_`, `has_`, or similar.

```suggestion
                    is_deletable, warning = filter(pod)
```